src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -120,15 +121,19 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
 
     private fun checkMethodCallExpr(holder: AnnotationHolder, o: RsMethodCall) {
         val fn = o.reference.resolve() as? RsFunction ?: return
-        if (fn.isUnsafe) {
+        val isExternalDecl = fn.parent as? RsForeignModItem != null
+
+        if (fn.isUnsafe || isExternalDecl) {
             checkUnsafeCall(holder, o.parentDotExpr)
         }
     }
 
     private fun checkCallExpr(holder: AnnotationHolder, o: RsCallExpr) {
         val path = (o.expr as? RsPathExpr)?.path ?: return
         val fn = path.reference.resolve() as? RsFunction ?: return
-        if (fn.isUnsafe) {
+        val isExternalDecl = fn.parent as? RsForeignModItem != null
+
+        if (fn.isUnsafe || isExternalDecl) {
```

Comment:
Looks like a code duplication
But if you move this logic into `isUnsafe` property, it will be fixed automatically